### PR TITLE
Remove client_for

### PR DESCRIPTION
**Pull request by breerly**
- [x] remove `tchannel.sync.thrift.client_for`
- [x] remove `tchannel.thrift.client_for`
- [x] update docs
- [x] changelog

Needed for #258.

### CODE REVIEW

File: tests/schemes/test_thrift.py [hunk 1]
```diff
@@ -1079,55 +1074,6 @@ def test_headers_should_be_a_map_of_strings(headers, service):
 
 @pytest.mark.gen_test
 @pytest.mark.call
-@pytest.mark.parametrize('ClientTChannel', [TChannel, DeprecatedTChannel])
```
Comment (abhinav) [inline tests/schemes/test_thrift.py]: I think you'll have to modify the fixtures here too. The fixtures tried all 4 combinations of thrift and thriftrw.

Comment (abhinav) [inline tests/schemes/test_thrift.py]: Although I don't think they used client_for, only thrift_request_builder

File: tests/testing/vcr/test_server.py [hunk 2]
```diff
@@ -25,7 +25,6 @@
 
 from contextlib2 import contextmanager
 
-# from tchannel.thrift import client_for
 from tchannel.errors import TChannelError
```
Comment (abhinav) [inline tests/testing/vcr/test_server.py]: delete comment

**Comment by breerly** [comment]
@abhinav @junchaowu @blampe

File: tests/schemes/test_thrift.py (hunk 1 above)
Comment (breerly) [inline tests/schemes/test_thrift.py]: Both done.

**Comment by junchaowu** [comment]
can you check the docs

**Comment by abhinav** [comment]
:+1:

**Comment by breerly** [comment]
@junchaowu sure did - lookin good!

**Comment by blampe** [comment]
it's a bold move, cotton, let's see if it pays off. :+1:

**Pull request merged**